doc/tutorial_mapping.md

```diff
@@ -0,0 +1,681 @@
+    }
+  }
+]
+
+```
+Let's go through that line-by-line.
+> I'm not going to help you too much with the JSON, I'm assuming you can follow along or read another guide. Basically, everything but numbers is enclosed in quotes, and you need a comma at the end of each item unless you're going to put in a closing } or ]. From there you can either blunder your way through like I did or get some help elsewhere. Clean up your code in http://dev.narc.ro/cataclysm/format.html to get it looking pretty, and if that says something useless like "syntax error" you could try www.jsonlint.com to tell you what error you actually have.
+
+- **"type": "mapgen", "method": "json",** = This is a mapgen file in JSON. Tells the game how to read it. Critical.
+- **"om_terrain":** = In this space you'll tell the game what this map is called on the overmap. Later on I'll show you how to make the overmap spawn list call on the map, based on the name you give it here.
+- **"object":** = Everything between the upcoming { and closing } is going to be the map file itself. Everything before this entry is collectively all metadata about the map, ie. ways to tell the game how to handle it.
+- **"fill_ter": "TK",** = "fill terrain". This is a huge timesaving measure that makes your maps much more robust. This tells the game "if I didn't tell you what kind of terrain to put in this tile, use the fill terrain". If you have a map with interiors, make sure this is whatever the most common floor you use is. Otherwise, just make it whatever the most common terrain type on your map is.
+- **"rows": [** = This is the map itself. Currently our map is a 24x24 blank area. Everything between those [ ] is going to be a map.
+- **"terrain": { "TK": "TK_id" }, "furniture": { "TK": "TK_id" }** = These are going to tell the game what our letters and numbers and ~s mean. Let's define those now.
+
+### Defining Terrain and Furniture
+We're going to be making a map that goes out in the field, so our main terrain types are going to be t_dirt, t_grass, t_grass_long, and t_grass_tall. We're going to make a rough circle of long and tall grass surrounded by dirt and short grass, with scattered rocks here and there.
+
+Before we get going let's name our map "tall_grass_patch".
+```
+"om_terrain": [ "tall_grass_patch" ],
+```
+
+Let's start with some easy definitions.
+```
+"terrain": {
+  " ": "t_dirt",
+  ".": "t_grass",
+  ",": "t_grass_long",
+  "|": "t_grass_tall"
+}
+```
+There. Now when I put a space on the mapgen file, it will make a patch of dirt. A period will make regular grass, a comma will make long grass, and a bar will make tall grass.
+
+> **How do I know what terrains to use, this file is thousands of lines long?** Well, there are two ways. You can either use *search* in the file to look for something, if you already know that terrain exists in-game; or, you can open up another mapgen file from the game folder and see what some of the terrains used in it are. I recommend both methods.
```

Comment:
In this statement you mention "this file" and "the file" but I don't know what your referring to.  What file?